js/classes/game_engine.js

```diff
@@ -102,105 +139,56 @@ class GameEngine {
      * This method is going to clear the canvas and redraw ALL of the entities in their *new* positions.
      */
     draw() {
-        // Clear the whole canvas with transparent color (rgba(0, 0, 0, 0))
-        CTX.clearRect(0, 0, CANVAS.width, CANVAS.height);
+        // (1) Paint over everything with the background color.
         CTX.fillStyle = BG_COLOR;
         CTX.fillRect(0, 0, Camera.SIZE.x, Camera.SIZE.y);
-        // Draw entities from first to last.
-        for (let i = 0; i < this.entities.length; i++) {
-            this.entities[i].draw();
-        }
-        // If we're debugging, draw the grid.
+
+        // (2) Draw the background entities.
+        this.entities.background.forEach((entity) => {entity.draw();});
+
+        // (3) Draw the midground entities.
+        this.entities.midground.forEach((entity) => {entity.draw();});
+
+        // (4) Draw CHAD and the debugging grid:
         if (this.debug) {
             this.drawGrid();
         }
-        // Draw Chad, who is not a regular entity.
         CHAD.draw();
+
+        // (5) Draw the foreground entities.
+        this.entities.foreground.forEach((entity) => {entity.draw();});
     };
 
     /**
      * This method is going to start listening for user inputs. Affects the Game Engine's left/right/up/down booleans according
      * to interaction with either the WASD keys or arrows.
      */
-    startInput() {
-
-        CANVAS.addEventListener("mousedown", (e) => {
-            this.user.firing = false;
-            this.user.aiming = true;
+    configureEventListeners() {
+        Object.keys(GAME.user).forEach((key) => {
```

Comment:
Great work cleaning this up!

Reply:
Not just cleaning; this was necessary in order to be able to reconfigure the CANVAS's listeners without reinitializing it and the context (which might have been the way to go, but screw it, what's done is done)